core/platform/search/elastic_search_services.py

```diff
@@ -0,0 +1,167 @@
+            Every key in the document is a field name, and the corresponding
+            value will be the field's value.
+            There MUST be a key named 'id', its value will be used as the
+            document's id.
+        index_name: str. The name of the index to insert the document into.
+
+    Raises:
+        Exception. A document cannot be added to the index.
+    """
+    assert isinstance(index_name, python_utils.BASESTRING)
+
+    for document in documents:
+        assert 'id' in document
+    for document in documents:
+        response = ES.index(
+            index_name, document, id=document['id'])
+        if response is None or response['_shards']['failed'] > 0:
+            raise Exception(
+                'Failed to add document to index.')
+
+
+def delete_documents_from_index(doc_ids, index_name):
+    """Deletes documents from an index.
+
+    Args:
+        doc_ids: list(str). A list of document ids of documents to be deleted
+            from the index.
+        index_name: str. The name of the index to delete the document from.
+
+    Raises:
+        Exception. Document id does not exist.
+    """
+    assert isinstance(index_name, python_utils.BASESTRING)
+    for doc_id in doc_ids:
+        assert isinstance(doc_id, python_utils.BASESTRING)
+
+    for doc_id in doc_ids:
+        if ES.exists(index_name, doc_id):
+            ES.delete(index_name, doc_id)
+        else:
+            raise Exception(
+                'Document id does not exist: %s' % doc_id)
+
+
+def clear_index(index_name):
+    """Clears an index on the elastic search instance.
+
+    Args:
+        index_name: str. The name of the index to clear.
+    """
+    assert isinstance(index_name, python_utils.BASESTRING)
+    # More details on clearing an index can be found here:
+    # https://elasticsearch-py.readthedocs.io/en/master/api.html#elasticsearch.Elasticsearch.delete_by_query
+    ES.delete_by_query(
+        index_name,
+        {
+            'query':
+                {
+                    'match_all': {}
+                }
+        })
+
+
+def search(
+        query_string, index_name, offset=None,
+        limit=feconf.SEARCH_RESULTS_PAGE_SIZE, ids_only=False):
+    """Searches for documents matching the given query in the given index.
+    NOTE: We cannot search through more than 10,000 pages from a search by
+    paginating using limit and offset. If the number of items to search through
+    is greater that 10,000, use the elasticsearch scroll API instead.
+
+    Args:
+        query_string: str. The search definition using Query DSL.
+        index_name: str. The name of the index. Use '_all' or empty string to
+            perform the operation on all indices.
+        offset: int. The offset into the index. Pass this in to start at the
+            'offset' when searching through a list of results of max length
+            'limit'. Leave as None to start at the beginning.
```

Comment:
This is still stated confusingly. Here, 'limit' sounds like the total number of individual results that match the search query (since you're "searching through" those results), but the parameter's default value is set to "search results page size" which seems wrong conceptually. Could you please clarify?

Reply:
I changed the API to use size instead of limit since that makes more sense and adheres to the elastic search API, which also uses size.